### PR TITLE
Feat/spotify autocomplete

### LINKED ISSUE #262: Spotify Intergration: Set-up Authentication

### PR DESCRIPTION
**Pull request by rensdoodeman**
Closes #262 and #22 

Also implemented debouncing but that's not in the video.

https://github.com/ASVGay/the-rhapsodies/assets/99396963/8523c149-a6d9-4530-933e-f4605e032001

### CODE REVIEW

File: cypress/e2e/new-suggestion/new-suggestion-song-information.cy.ts [hunk 1]
```diff
@@ -2,8 +2,9 @@ import { updateNewSuggestion } from "@/redux/slices/new-suggestion.slice"
 import {
   shouldGoToInstrumentsArea,
   newSuggestionFilledSongInformation,
-  shouldContainJSONSongInformationInState,
+  shouldContainJSONSongInformationInState
 } from "./helpers/new-suggestion.helper"
+import { wait } from "next/dist/build/output/log"
```
Comment (fkoulen) [inline cypress/e2e/new-suggestion/new-suggestion-song-information.cy.ts:7]: This is prob a lil mistake right?

Comment (rensdoodeman) [reply cypress/e2e/new-suggestion/new-suggestion-song-information.cy.ts:7]: oop- yes

File: src/components/suggestion/areas/song-information.area.tsx [hunk 2]
```diff
@@ -30,6 +60,43 @@ const SongInformationArea = ({ proceedToNextArea, onFormSuccess }: SongInformati
...
+        }))
+        setSearchResults(items)
+      })
+      .catch(() => setShowSearchError(true))
+      .finally(() => setFetchingSongs(false))
+  }, 400)
+
+  const handleSearch = (value: string) => {
+    setValue("title", value)
+    debouncedHandleSearch(value)
```
Comment (fkoulen) [inline src/components/suggestion/areas/song-information.area.tsx:84]: Did `debounce(handleSearch(value), 400)` not work?

File: src/components/suggestion/areas/song-information.area.tsx [hunk 3]
```diff
@@ -50,25 +117,61 @@ const SongInformationArea = ({ proceedToNextArea, onFormSuccess }: SongInformati
...
+                placeholder="Search for a song title"
+                className={`${errors.title && "error"}`}
+                {...register("title", {
+                  validate: (value) => !!value.trim()
+                })}
+                onFocus={() => setIsSearchFocused(true)}
+                onBlur={handleSearchBlur}
+                onChange={(event) => {
+                  handleSearch(event.target.value)
+                }}
```
Comment (fkoulen) [inline src/components/suggestion/areas/song-information.area.tsx:149]: Could possibly be a one liner like the lines before this but is not necessary!

**Comment by fkoulen** [comment]
It looks very good. I'm hella impressed!!! If you could just fix the tests I'll throw an approve on it anytime! ❤️

File: src/components/suggestion/areas/song-information.area.tsx (hunk 2 above)
Comment (rensdoodeman) [reply src/components/suggestion/areas/song-information.area.tsx:84]: no bc i need the `event` from the onChange and `(event) => debounce(..` doesnt work it, you have to call debounce right away like `onChange={debounce(... ` or `const example = debounce(..`

**Review by fkoulen — APPROVED** [review]

**Pull request merged**